components/utils/args.py

```diff
@@ -41,3 +59,328 @@ def add_arguments_sorting(a_arg_parser):
     return a_arg_parser
+
+# Result sorting related arguments
+arg_sort_field = {
+        "arg" : "--sort_field",
+        "default" : "",
+        "help" : "The optional column name to sort a query result list by.",
+        "required" : False
+    }
+
+arg_sort_order = {
+        "arg" : "--sort_order",
+        "default" : "+",
+        "help" : "The optional direction to sort a query result list by.",
+        "required" : False
+    }
+
+# File and folder related arguments. Both files and folders are represented in the same way but different parameter names
+# are provided so as to improve code comprehension.
+arg_file_name = {
+        "arg" : "--file_name",
+        "default" : str(uuid.uuid4()),
+        "help" : "The name of the file in the file system.",
+        "required" : True
+    }
+
+arg_file_uuid = {
+        "arg" : "--file_uuid",
+        "default" : str(uuid.uuid4()),
+        "help" : "The UUID of the file as a string.",
+        "required" : True
+    }
+
+arg_file_parent_uuid = {
+        "arg" : "--file_parent_uuid",
+        "default" : None,
+        "help" : "The UUID of the file's parent (if avaialble) as a string. This represents a containing folder.",
+        "required" : False
+    }
+
+arg_folder_name = {
+        "arg" : "--folder_name",
+        "default" : str(uuid.uuid4()),
+        "help" : "The name of the folder in the file system.",
+        "required" : True
+    }
+
+arg_folder_uuid = {
+        "arg" : "--folder_uuid",
+        "default" : str(uuid.uuid4()),
+        "help" : "The UUID of the folder as a string.",
+        "required" : True
+    }
+
+arg_folder_parent_uuid = {
+        "arg" : "--folder_parent_uuid",
+        "default" : None,
+        "help" : "The UUID of the folder's parent (if avaialble) as a string. This represents a containing folder.",
+        "required" : False
+    }
+
+# Site and organization related arguments
+arg_site_name = {
+        "arg" : "--site_name",
+        "default" : "Site",
+        "help" : "A name to refer to the site. This name is visible to operators for selection on client software and in the web portal map.",
+        "required" : True
+    }
+
+arg_site_id = {
+        "arg" : "--site_id",
+        "default" : "",
+        "help" : "64 character site identifier.",
+        "required" : True
+    }
+
+arg_site_latitude = {
+        "arg" : "--site_latitude",
+        "default" : "",
+        "help" : "The latitude of an arbitrary point to locate the site on a map.",
+        "required" : True
+    }
+
+arg_site_longitude = {
+        "arg" : "--site_longitude",
+        "default" : "",
+        "help" : "The longitude of an arbitrary point to locate the site on a map.",
+        "required" : True
+    }
+
+arg_site_timezone = {
+        "arg" : "--site_timezone",
+        "default" : "",
+        "help" : "The timezone that the site is located in the format 'Australia/Brisbane' for the purpose of localizing dates and times in reports and other areas.",
```

Comment:
This might read:
"help" : "The IANA timezone that the site is located in (eg. 'Australia/Brisbane') for the purpose of localizing dates and times in reports and other areas.",

Reply:
Agreed.